roachtest: tpccbench/nodes=3/cpu=16 failed

SHA: https://github.com/cockroachdb/cockroach/commits/7dab0dcfd37c389af357c302c073b9611b5ada25

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=tpccbench/nodes=3/cpu=16 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1398203&tab=buildLog

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20190721-1398203/tpccbench/nodes=3/cpu=16/run_1
	cluster.go:2090,tpcc.go:842,tpcc.go:559,test_runner.go:691: unexpected node event: 3: dead

```

Node 3 [crashed](https://teamcity.cockroachdb.com/repository/download/Cockroach_Nightlies_RoachtestNightlyAws/1494403:id/tpccbench/nodes%3D3/cpu%3D16/3.logs/cockroach-rocksdb.ip-10-12-36-141.ubuntu.2019-09-18T07_28_07Z.021771.log) due to a nil pointer dereference:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1f449b6]

goroutine 19990 [running]:
panic(0x3e77e40, 0x75fac40)
	/usr/local/go/src/runtime/panic.go:565 +0x2c5 fp=0xc00a7f8718 sp=0xc00a7f8688 pc=0x78d565
runtime.panicmem(...)
	/usr/local/go/src/runtime/panic.go:82
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:390 +0x411 fp=0xc00a7f8748 sp=0xc00a7f8718 pc=0x7a30c1
github.com/cockroachdb/cockroach/pkg/sql/row.(*Fetcher).GetRangesInfo(0xc01a12f680, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/row/fetcher.go:1414 +0x26 fp=0xc00a7f8778 sp=0xc00a7f8748 pc=0x1f449b6
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*tableReader).generateMeta(0xc00f54f100, 0x4ce7c80, 0xc01d541c40, 0xcb1971, 0x4105760, 0x42205c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/tablereader.go:273 +0x410 fp=0xc00a7f8950 sp=0xc00a7f8778 pc=0x2053610
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*tableReader).generateTrailingMeta(0xc00f54f100, 0x4ce7c80, 0xc01d541c40, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/tablereader.go:144 +0x43 fp=0xc00a7f89a8 sp=0xc00a7f8950 pc=0x2052323
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*tableReader).generateTrailingMeta-fm(0x4ce7c80, 0xc01d541c40, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/tablereader.go:143 +0x3e fp=0xc00a7f89e8 sp=0xc00a7f89a8 pc=0x20651ae
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBase).moveToTrailingMeta(0xc00f54f100)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:687 +0x136 fp=0xc00a7f8ae8 sp=0xc00a7f89e8 pc=0x2004576
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBase).MoveToDraining(0xc00f54f100, 0x4c71b40, 0xc013dd8560)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:589 +0x195 fp=0xc00a7f8be0 sp=0xc00a7f8ae8 pc=0x2003f75
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*tableReader).Start(0xc00f54f100, 0x4ce7c80, 0xc01d541c40, 0xc000e0ffa8, 0x1b1e28e)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/tablereader.go:184 +0x35f fp=0xc00a7f8f20 sp=0xc00a7f8be0 pc=0x20526ef
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*samplerProcessor).Run(0xc0082e8a80, 0x4ce7c80, 0xc01d541c40)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/sampler.go:178 +0x5b fp=0xc00a7f8fa0 sp=0xc00a7f8f20 pc=0x203f59b
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).startInternal.func1(0xc01a12f560, 0xc019ac7380, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:284 +0x65 fp=0xc00a7f8fc8 sp=0xc00a7f8fa0 pc=0x201bc05
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1337 +0x1 fp=0xc00a7f8fd0 sp=0xc00a7f8fc8 pc=0x7bd2f1
created by github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).startInternal
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:283 +0x35b
```
@jordanlewis could someone on SQL execution take a look at this? It looks like an issue in `row.Fetcher`.

--- a/pkg/sql/row/fetcher.go
+++ b/pkg/sql/row/fetcher.go
@@ -1423,7 +1423,7 @@ func (rf *Fetcher) PartialKey(nCols int) (roachpb.Key, error) {
 // GetRangesInfo returns information about the ranges where the rows came from.
 // The RangeInfo's are deduped and not ordered.
 func (rf *Fetcher) GetRangesInfo() []roachpb.RangeInfo {
-	f := rf.kvFetcher.kvBatchFetcher
+	f := rf.kvFetcher
 	if f == nil {
 		// Not yet initialized.
 		return nil
@@ -1433,7 +1433,12 @@ func (rf *Fetcher) GetRangesInfo() []roachpb.RangeInfo {
 
 // GetBytesRead returns total number of bytes read by the underlying KVFetcher.
 func (rf *Fetcher) GetBytesRead() int64 {
-	return rf.kvFetcher.bytesRead
+	f := rf.kvFetcher
+	if f == nil {
+		// Not yet initialized.
+		return 0
+	}
+	return f.bytesRead
 }
 
 // Only unique secondary indexes have extra columns to decode (namely the

--- a/pkg/sql/row/fetcher_test.go
+++ b/pkg/sql/row/fetcher_test.go
@@ -27,6 +27,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/stretchr/testify/assert"
 )
 
 type initFetcherArgs struct {
@@ -1051,4 +1052,13 @@ func TestRowFetcherReset(t *testing.T) {
 
 func idLookupKey(tableID TableID, indexID sqlbase.IndexID) uint64 {
 	return (uint64(tableID) << 32) | uint64(indexID)
+}
+
+func TestFetcherUninitialized(t *testing.T) {
+	// Regression test for #39013: make sure it's okay to call GetRangesInfo and
+	// GetBytesReader even before the fetcher was fully initialized.
+	var fetcher Fetcher
+
+	assert.Nil(t, fetcher.GetRangesInfo())
+	assert.Zero(t, fetcher.GetBytesRead())
 }

